src/relax/ir/expr.cc

```diff
@@ -499,49 +484,35 @@ Function::Function(Array<Var> params, Expr body, Type ret_type, Expr ret_shape,
 }
 
 TVM_REGISTER_GLOBAL("relax.Function")
-    .set_body_typed([](Array<Var> params, Expr body, Type ret_type, Expr ret_shape, DictAttrs attrs,
-                       Span span) {
-      return Function(params, body, ret_type, ret_shape, attrs, span);
-    });
-
-Function Function::CreateUnchecked(Array<Var> params, Expr body, Type ret_type, Expr ret_shape,
-                                   DictAttrs attrs, Span span) {
-  // TODO(@Hzfengsy): revisit `CreateUnchecked` after the parser_v1 removed
+    .set_body_typed([](Array<Var> params, Expr body, Optional<StructInfo> ret_struct_info,
+                       DictAttrs attrs,
+                       Span span) { return Function(params, body, ret_struct_info, attrs, span); });
 
+Function Function::CreateEmpty(Array<Var> params, StructInfo ret_struct_info, DictAttrs attrs,
+                               Span span) {
   Array<StructInfo> param_sinfo;
-
-  for (Var param : params) {
+  for (const Var& param : params) {
     ICHECK(param->checked_type_.defined())
         << "relax.Function requires params to contain checked_type_.";
     param_sinfo.push_back(GetStructInfo(param));
   }
-
-  StructInfo ret_info;
-
-  if (ret_type.defined()) {
-    ret_info = StructInfoFromTypeLegacyShapeHint(ret_type, ret_shape);
-  } else {
-    ret_info = FuncStructInfo::OpaqueFunc();
-  }
-  FuncStructInfo finfo(param_sinfo, ret_info);
+  FuncStructInfo finfo(param_sinfo, ret_struct_info);
 
   // set the fields
   ObjectPtr<FunctionNode> n = make_object<FunctionNode>();
   n->params = std::move(params);
-  n->body = std::move(body);
+  n->body = Expr();
   n->checked_type_ = GetStaticType(finfo);
   n->struct_info_ = std::move(finfo);
-  n->ret_type = std::move(ret_type);
-  n->ret_shape = std::move(ret_shape);
+  n->ret_struct_info = std::move(ret_struct_info);
   n->attrs = std::move(attrs);
   n->span = std::move(span);
   return Function(std::move(n));
 }
 
-TVM_REGISTER_GLOBAL("relax.Function_CreateUnchecked")
-    .set_body_typed([](Array<Var> params, Expr body, Type ret_type, Expr ret_shape, DictAttrs attrs,
-                       Span span) {
-      return Function::CreateUnchecked(params, body, ret_type, ret_shape, attrs, span);
+TVM_REGISTER_GLOBAL("relax.Function_CreateEmpty")
```

Comment:
🤔 (though unrelated to the change of this PR)
```suggestion
TVM_REGISTER_GLOBAL("relax.FunctionCreateEmpty")
```

Reply:
updated